src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +106,49 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _startRequestUri);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _startRequestUri);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _startRequestUri);
                     return;
             }
         }
 
+        internal static string AppendOrReplaceApiVersion(string uri, Uri startRequestUri)
+        {
+            const string apiVersionQuery = "api-version";
+            NameValueCollection startRequestUriQueries = HttpUtility.ParseQueryString(startRequestUri.Query);
```

Comment:
Given this is on the hotpath I think we can improve this more.  It appears that you are constructing a collection which will contain all of the query params.  This is a lot of allocation and time that isn't needed since all we want is the apiVersion query param.

As an example here is the benchmark diff
|                     Method |      Mean |    Error |   StdDev |  Gen 0 | Allocated |
|--------------------------- |----------:|---------:|---------:|-------:|----------:|
| GetApiStringWithParseQuery | 981.35 ns | 7.970 ns | 7.065 ns | 0.3529 |   2,216 B |
|       GetApiStringWithSpan |  28.57 ns | 0.254 ns | 0.225 ns | 0.0102 |      64 B |

The ParseQueryString is 34x slower and allocates 34x more vs doing a simple parse of this one value out of the string.
Here is the code I used to test

```csharp
[MemoryDiagnoser]
public class ParseQueryBench
{
    private const string ApiVersionParam = "api-version";
    private static Uri StartRequestUri = new Uri("https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRG-416/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testRi-6086?something=something&api-version=2021-09-30-PREVIEW&foobar=foobar");

    [Benchmark]
    public string GetApiStringWithParseQuery()
    {
        NameValueCollection startRequestUriQueries = HttpUtility.ParseQueryString(StartRequestUri.Query);
        return startRequestUriQueries.Get(ApiVersionParam);
    }

    [Benchmark]
    public string GetApiStringWithSpan()
    {
        ReadOnlySpan<char> uriSpan = StartRequestUri.Query.AsSpan();
        int startIndex = uriSpan.IndexOf(ApiVersionParam);
        startIndex += ApiVersionParam.Length + 1;
        ReadOnlySpan<char> remaining = uriSpan.Slice(startIndex);
        int endIndex = remaining.IndexOf('&');
        int length = endIndex == -1 ? uriSpan.Length - startIndex : endIndex;
        return uriSpan.Slice(startIndex, length).ToString();
    }
}
```

I didn't find any utilities in dotnet code that does the same thing so I think we should use this and just add a robust set of tests around it to make sure it works in all cases.